Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1592,6 +1593,26 @@ void RenderLayer::setAncestorChainHasVisibleDescendant()
 
 void RenderLayer::updateAncestorDependentState()
 {
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    m_enclosingSVGResourceContainer = nullptr;
+    m_enclosingSVGHiddenContainer = nullptr;
+    auto determineSVGAncestors = [&] (const RenderElement& renderer) {
+        for (auto* ancestor = renderer.parent(); ancestor; ancestor = ancestor->parent()) {
```

Comment:
We started at 'renderer' before, not 'renderer.parent()' when we used lineageOfType. However, I do think this version is more correct.